formats/datatables/resources/ext.srf.formats.datatables.js

```diff
@@ -595,6 +595,23 @@
 				},
 			});
 
+			// cacheKey ensures that the cached pages
+			// are related to current sorting and searchPanes filters
+			var getCacheKey = function (obj) {
+				return (
+					JSON.stringify(obj.order) +
+					(!searchPanes
+						? ""
+						: JSON.stringify(
+								Object.keys(obj.searchPanes).length
+									? obj.searchPanes
+									: Object.fromEntries(
+											Object.keys(columnDefs).map((x) => [x, {}])
+									  )
+						  ))
+				);
+			};
```

Comment:
Looks like you could benefit from extracting small dedicated functions. There currently is a lot of indentation and the scope is large.

Reply:
This PR is a little step in that direction